csp/socket.csp

```diff
@@ -0,0 +1,170 @@
+<csp:class super="%CSP.WebSocket" abstract="1">
+<script method="Run" language="cache" runat="server">
+    If %request.Get("method") = "preview" {
+        Do ##class(SourceControl.Git.API).Pull(1)
+    } ElseIf %request.Get("method") = "pull" {
+        Do ##class(SourceControl.Git.API).Pull()
+    } ElseIf %request.Get("method") = "init" {
+        Do ##class(SourceControl.Git.Utils).Init()
+		Write !,"Done."
+    } ElseIf %request.Get("method") = "clone" {
+        Set remote = %request.Get("remote")
+        Do ##class(SourceControl.Git.Utils).Clone(remote)
+		Write !,"Done."
+    } ElseIf %request.Get("method") = "sshkeygen" {
+        Do ##class(SourceControl.Git.Utils).GenerateSSHKeyPair()
+		Write !,"Done."
+    } Else {
+        Write !!,"Invalid method selected.",!!
+    }
+</script>
+<script method="OnPreServer" language="cache" returntype="%Status" runat="server" classmethod="##(+$Piece($STACK($STACK-3,"PLACE"),"+",2)<10)##">
```

Comment:
I'd really rather not do this... probably just need to bite the bullet and add a %z class to the package.